populator-machinery/controller_test.go

```diff
@@ -62,19 +63,30 @@ type testCase struct {
 	populateFn func(context.Context, PopulatorParams) error
 	// Provider specific data population completeness check function, return true when data transfer gets completed.
 	populateCompleteFn func(context.Context, PopulatorParams) (bool, error)
+	// The original PVC gets deleted or not
+	pvcDeleted bool
 	// PvcPrimeMutator is the mutator function for pvcPrime
 	pvcPrimeMutator func(PvcPrimeMutatorParams) (*v1.PersistentVolumeClaim, error)
 	// Expected errors
 	expectedResult error
 	// Expected keys in the notifyMap
 	expectedKeys []string
 	// Expected objects after the test runs
-	expectedObjects vpObjects
+	expectedObjects *vpObjects
 }
 
 // vpObjects includes the objects we want to compare after tests run
 type vpObjects struct {
-	pvcPrime *corev1.PersistentVolumeClaim
+	pvc *corev1.PersistentVolumeClaim
+	// When set to true, only compare PVC Finalizers
+	onlyComparePVCFinalizers bool
+	pvcPrime                 *corev1.PersistentVolumeClaim
+	// When set to true, only compare mutate feilds
+	onlyComparePVCPrimeMutateFields bool
+	pod                             *v1.Pod
+	// When set to true, only verify Pod not nil
+	onlyCheckPodExists bool
```

Comment:
I don't see this field being set at all in test cases. Is it needed?

Reply:
Removed